### PR TITLE
Edits and such

### PR DESCRIPTION
**Pull request by itamarhaber**
Great addition - sorry for all the formatting changes.

### CODE REVIEW

File: topics/rediscli.md [hunk 1]
```diff
@@ -94,27 +92,27 @@ option:
...
 ## Getting input from other programs
 
-There are two ways you can use `redis-cli` in order to get the input
-from other commands (from the standard input, basically).
-One is to use as last argument the payload we read from *stdin*.
-For example in order to set a Redis key to the content of the
-file `/etc/services` if my computer, I can use the `-x` option:
+There are two ways you can use `redis-cli` in order to get the input from other
+commands (from the standard input, basically). One is to use as last argument
+as the payload we read from *stdin*. For example, in order to set a Redis key
```
Comment (antirez) [inline topics/rediscli.md]: show this "as" be "the"?

Comment (itamarhaber) [inline topics/rediscli.md]: Yep.

File: topics/rediscli.md [hunk 2]
```diff
@@ -229,12 +227,12 @@ This is very useful for scripts and certain types of testing, however most
...
 
-In interactive mode the user types Redis commands into a prompt. The command
+In interactive mode the user types Redis commands at the prompt. The command
 is sent to the server, processed, and the reply is parsed back and rendered
-in a simpler to read form.
+into a simpler form to read.
 
-In order to run the CLI in interactive mode, there is nothing special to do:
-just lunch it without arguments and you are in:
+Nothing special is needed for runung the CLI in interactive mode -
```
Comment (antirez) [inline topics/rediscli.md]: runung -> running?

Comment (itamarhaber) [inline topics/rediscli.md]: Tep :smile:

File: topics/rediscli.md [hunk 3]
```diff
@@ -442,13 +445,13 @@ and produces a quite verbose output:
     0 zsets with 0 members (00.00% of keys, avg size 0.00)
 
 In the first part of the output, each new key larger than the previous larger
-key (of the same type) encountered is reported. The summary section instead
+key (of the same type) encountered is reported. The summary section
 provides general stats about the data inside the Redis instance.
 
 The program uses the `SCAN` command, so it can be executed against a busy
-server without impacting the operations, however the `-i` option can be
+server without impacting the operations, but the `-i` option can be
```
Comment (antirez) [inline topics/rediscli.md]: makes more sense to take *however* here perhaps?

File: topics/rediscli.md [hunk 4]
```diff
@@ -507,12 +510,12 @@ kind of objects, by key name:
...
 The CLI is able to publish messages in Redis Pub/Sub channels just using
 the `PUBLISH` command. This is expected since the `PUBLISH` command is very
-similar to any other command. Different is the case of subscribing to channels
-in order to receive messages, in this case there is to block and wait for
-messages, so this is implemented as a special mode into `redis-cli`, however
-the mode is not enabled by using a special option, but simply by using the
-`SUBSCRIBE` or `PSUBSCRIBE` command, both in interactive or non interactive
-mode:
+similar to any other command. Subscribing to channels in order to receive
+is different messages - in this case there is to block and wait for
```
Comment (antirez) [inline topics/rediscli.md]: "is different messages" looks strange or I'm missing something?

File: topics/rediscli.md (hunk 3 above)
Comment (itamarhaber) [inline topics/rediscli.md]: Dunno. However, reverting.

File: topics/rediscli.md [hunk 5]
```diff
@@ -544,47 +547,49 @@ by a Redis instance:
...
+However this is usually not a big problem, since we are interested in
 events of a few millisecond or more.
 
 Sometimes it is useful to study how the maximum and average latencies
-evolve during time. For this goal the `--latency-history` option is used:
-it works exactly like `--latency`, but each 15 second (by default) a new
-sampling session is started, starting from zero:
+evolve during time. The `--latency-history` option is used for that
+purpose: it works exactly like `--latency`, but every 15 seconds (by
+default) a new sampling session is started beginning with zero:
```
Comment (antirez) [inline topics/rediscli.md]: I don't know why I wrote "starting from zero", the right thing should be "starting from scratch". What I mean is that it's a new sampling.

File: topics/rediscli.md (hunk 4 above)
Comment (itamarhaber) [inline topics/rediscli.md]: I guess I'm the one who missed something - thanks, fixing.

File: topics/rediscli.md [hunk 6]
```diff
@@ -684,27 +689,27 @@ in order to improve the bug report.
...
-tool was built: the main motivation was to test the Redis LRU implementation
-quality, but now is also useful in order to test how a given version behaves
-with the settings you had in mind for your deployment.
+Theoretically, given the distribution of the requests and the Redis memory
+overhead, it should be possible to compute the hit rate analytically with
+with a mathematical formula. However, Redis can be configured with
+different LRU settings (number of samples) and LRU's implementation, which
+is approximated in Redis, changes a lot between different versions. Similarly
+the amount of memory per key may change between versions. That is why this
+tool was built: its main was motivation for testing the quality of Redis' LRU
```
Comment (antirez) [inline topics/rediscli.md]: "its main was motivation"? probably typo for "its main motivation was"

**Comment by antirez** [comment]
Great work! I commented the diff with possible errors I spotted in your changes.

File: topics/rediscli.md (hunk 6 above)
Comment (itamarhaber) [inline topics/rediscli.md]: Lucky guess :smirk:

**Comment by itamarhaber** [comment]
ty & I admit to having now finally learned the fine intricacies of the `-i` option in all of its contexts :+1:

**Comment by antirez** [comment]
Thanks Itamar! Merged.

**Pull request merged**